### PR TITLE
completed challenge 4

### PR DESCRIPTION
**Pull request by MScott1118**

### CODE REVIEW

File: db/controllers.js [hunk 1]
```diff
@@ -8,7 +8,19 @@ exports.getTopics = (req, res) => {
...
-    // console.log(returnedEndpoints);
     return res.status(200).send({ endpoints: returnedEndpoints });
   });
 };
+
+exports.getArticleByID = (req, res, next) => {
+  const { article_id } = req.params;
+  return fetchArticleByID(article_id)
+    .then((returnedArticle) => {
+      if (returnedArticle.length === 0) {
```
Comment (two-triangles) [inline db/controllers.js]: This isn't the job of the controller, if you are going to reject, it should be done in the model and that rejected error would be captured by the catch block of the controller and passed on to the error handling middleware. 

Controller - only deals with request and response lifecycle 
Model - responsible for interacting with the database and dealing with the data

File: __tests__/app.test.js [hunk 2]
```diff
@@ -41,3 +44,27 @@ describe("GET API", () => {
...
+        expect(typeof res.body.article[0].created_at).toBe("string");
+        expect(typeof res.body.article[0].votes).toBe("number");
+        expect(typeof res.body.article[0].article_img_url).toBe("string");
+        expect(typeof res.body.article[0].article_id).toBe("number");
+      });
+  });
+  test("should return 400 error when passed an article id that doesn't exist", () => {
+    return request(app).get("/api/articles/40").expect(400);
+  });
+});
```
Comment (two-triangles) [inline __tests__/app.test.js:70]: This should be a 404 - not a 400. It is a perfectly valid article_id, it just doesn't exist yet. 400 designates that this is a bad request, of which it isn't. 404 - Not Found, is the way we want to go

File: db/models.js [hunk 3]
```diff
@@ -12,3 +12,14 @@ exports.fetchAPI = () => {
     res(endpoints);
   });
 };
+
+exports.fetchArticleByID = (article_id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    .then((returnedArticle) => {
+      if (returnedArticle.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Invalid article ID" });
```
Comment (Pipe-Smoking-Rabbit) [inline db/models.js:21]: Nice model great handling for if the rows length is 0

File: db/app.js [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
 const app = express();
 app.use(express.json());
 
 app.get("/api/topics", getTopics);
 
 app.get("/api", getAPI);
 
+app.get("/api/articles/:article_id", getArticleByID);
+app.use((err, req, res, next) => {
+  console.log(err);
```
Comment (Pipe-Smoking-Rabbit) [inline db/app.js:12]: Careful with leaving console logs in error handlers, as you develop more tests this log will fire many times when running the test suite

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Looks great! 👍

**Pull request merged**